wasi/2023/WASI-09-21.md

```diff
@@ -30,3 +30,180 @@ The meeting will be on a zoom.us video conference.
+
+**Pat Hickey:** Implementations are ongoing but we are no longer seeing major changes. Mostly smaller details that are being fleshed out in several proposals. Please be involved and understanding how this will impact you and your platform. The early we can talk about that, the better it is for everyone.
+
+**Pat Hickey:** Preview 2 is not our end goal. It is our ticket to keep working on this. We have to ship preview to be in the hands of our customers and be stable code that we can start building on it.
+
+**Petr Penzin**: On one-hand, let’s vote all at once. The vote from unanimous consent is a little different from what the CG does. The fact that we will move all of the proposals all at once. It is difficult to move everything all at once. Very different from what I’m used to doing within the W3C. 
+
+**Pat Hickey:** Vote early and often advice should help cover this.
+
+**Luke Wagner**: We have to be individually happy with each of these proposals to phase 3, then we will vote on a snapshot of the collection of these.
+
+**Bailey Hayes**: (chat) Several recent Wasm CG groups votes have been unanimous consent.
+
+**Piotr Sikora**: The goal stated in the slides was to vote everything at once to phase 3. 
+
+**Pat Hickey:** That was my goal. We can phase 3 vote ahead of time if necessary. It’s a lot of procedural votes and was hoping to reduce that. The proposals are very interrelated. Many of these like streams impact http and files. Many of the lower-level proposals were influenced by learning from the implementation.
+
+**Petr Penzin**: Now we as a WASI Subgroup we now we need to go around and move this all at once. It’s a very different dynamic.
+
+**Petr Penzin**: I actually have another question. You are using wasmtime and JCO transpile. One objection to this is that JCO is not really a runtime. JCO is a compatibility layer. Implementing something like that is different from having two different WebAssembly engines.
+
+**Pat Hickey:** The goal of a component runtime is to be able to run a component. Being able to run in a browser engine and in wasmtime means we were able to ship without needing native support. If the goal is only to be able to run it, then if we can run it. 
+
+**Petr Penzin**: wasm2c is not considered an engine.
+
+**Pat Hickey:** wasm2c is not a Web engine. The core wasm cg has a requirement for two Web engines, but the WASI subgroup does not use that requirement because we are concerned with use cases beyond just the Web.
+
+**Petr Penzin**: If you have two transpilers, then you could move it forward right?
+
+**Pat Hickey:** Yes
+
+**Luke Wagner**: If it is part of the portability requirements for the proposal as in those are the goals where that proposals needs to run, then yes.
+
+**Piotr Sikora**: Concern of underspecified specification since JCO and wasmtime are primarily developed by Fastly.
```

Comment:
@PiotrSikora I thought you asked if the two implementations were actually independent? Though this wording might be equivalent.

Reply:
Yes, that was my question, and I think this summary is mostly equivalent, but I agree that the intent behind it might not be clear to someone who didn't participate in the meeting. Perhaps this is better?

```suggestion
**Piotr Sikora**: Concern of underspecified specification since both "independent" implementations (JCO and Wasmtime) are primarily developed by Fastly.
```

Also, I've just realized that @tschneidereit's answer is about Wasmtime as a whole and not specifically about WASI in Wasmtime, but it's not like we're not going to block preview 2 on a truly independent implementation.

Reply:
It's true that I talked about Wasmtime and JCO overall, and that for the WASI implementations Fastly is doing a larger share of the work on both. The key point holds though: they are developed by pretty separate sets of people, and with completely different approaches to the implementation, so they should very well serve the purpose of vetting independent implementability.